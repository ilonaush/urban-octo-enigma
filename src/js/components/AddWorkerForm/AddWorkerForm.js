import React, {Component} from 'react';
import "./AddWorkerForm.styl";
import Input from "../Input/Input";

export default class AddWorkerForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
            cat: {
                name: '',
                age: '',
                color: '',
                reason: '',
                location: '',
                health: ''
            }
        };
    }

    /**
     * handles input change
     * @param value
     * @param name
     */
    handleChange = (value, name) => {
        this.setState((state) => ({
            cat: {
                ...state.cat,
                [name]: value
            }
        }))
    };

    render() {
        const {cat} = this.state;
        const {handleSubmit} = this.props;
        return (
            <form className='addWorkerForm' onSubmit={(e) => handleSubmit(e, cat)}>
                <h3>Please share some info about the cat with us</h3>
                <Input id='name-input'
                       name='name'
                       placeholder='Name'
                       onChange={(event) => this.handleChange(event.target.value, event.target.name)}
                       value={cat.name}
                       required
                        />
                <Input name='age'
                       placeholder='Age'
                       onChange={(event) => this.handleChange(event.target.value, event.target.name)}
                       value={cat.age}
                       required
                />
                <Input name='color'
                       placeholder='Color'
                       onChange={(event) => this.handleChange(event.target.value, event.target.name)}
                       value={cat.color}
                       required
                />
                <textarea name='reason'
                       placeholder='Reason'
                       onChange={(event) => this.handleChange(event.target.value, event.target.name)}
                       value={cat.reason}
                />
                <Input name='location'
                       placeholder='Location'
                       onChange={(event) => this.handleChange(event.target.value, event.target.name)}
                       value={cat.location}
                       required
                />
                <Input name='health'
                       placeholder='Describe health'
                       onChange={(event) => this.handleChange(event.target.value, event.target.name)}
                       value={cat.health}
                       required
                />
                <button id='submit-btn' type='submit'>Accept the cat</button>
            </form>
        );
    }
}

AddWorkerForm.propTypes = {};

