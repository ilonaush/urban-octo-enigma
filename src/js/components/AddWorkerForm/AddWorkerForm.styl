.addWorkerForm
  display flex
  flex-direction column
