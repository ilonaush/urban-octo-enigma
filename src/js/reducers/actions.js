export default {
    employWorker: function (worker) {
        return {
            type: 'EMPLOY_WORKER',
            worker: worker
        }
    },
    editWorkTime: function (worker) {
        return {
            type: 'EDIT_WORK_TIME',
            worker: worker,
        }
    },
    fireWorker: function (workerId) {
        return {
            type: 'FIRE_WORKER',
            id: workerId
        }
    }
};
