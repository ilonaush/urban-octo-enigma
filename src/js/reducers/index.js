import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer   from './reducers';
import  { saveStore, loadStore } from '../services/LocalStorageService';

const predefinedStore = loadStore() || {};

const store = createStore(
    reducer,
    predefinedStore

);

store.subscribe(() => {
    saveStore(store)
});

export {
    store, predefinedStore
};
