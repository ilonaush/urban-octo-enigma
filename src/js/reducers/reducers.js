import {ACTIONS_TYPES} from "../constants";

const initialState = {
  loading: false,
};

let reducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS_TYPES.GET_CATS:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ADD_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.FEED_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.HUG_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.WASH_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ISSUE_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.SET_LOADING: {
      return {
        ...state,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      }
    }
    default:
      return state;
  }
};

export default reducer;
