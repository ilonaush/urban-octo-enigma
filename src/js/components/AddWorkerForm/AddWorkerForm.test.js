import React from 'react';

import AddWorkerForm  from './AddWorkerForm';

import ReactDOM from 'react-dom';
import {MemoryRouter} from "react-router";
import {Provider} from "react-redux";
import {store} from "../../reducers/index";


describe('AddWorkerForm', () => {
    it('renders', () => {

        const div = document.createElement('div');
        ReactDOM.render(<Provider store={store}><AddWorkerForm/></Provider>, div)
    })
})