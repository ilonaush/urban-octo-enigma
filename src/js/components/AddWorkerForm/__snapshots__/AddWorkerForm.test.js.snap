// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWorkerForm renders correctly 1`] = `
<form
  className="addWorkerForm"
  onSubmit={[Function]}
>
  <h3>
    Please share some info about the cat with us
  </h3>
  <input
    id="name-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    required={true}
    value=""
  />
  <input
    name="age"
    onChange={[Function]}
    placeholder="Age"
    required={true}
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    placeholder="Color"
    required={true}
    value=""
  />
  <textarea
    name="reason"
    onChange={[Function]}
    placeholder="Reason"
    value=""
  />
  <input
    name="location"
    onChange={[Function]}
    placeholder="Location"
    required={true}
    value=""
  />
  <input
    name="health"
    onChange={[Function]}
    placeholder="Describe health"
    required={true}
    value=""
  />
  <button
    id="submit-btn"
    type="submit"
  >
    Accept the cat
  </button>
</form>
`;
